### PR TITLE
Bilingual slug

### LINKED ISSUE #366: Translate results query string.
To comply with the spirit of the Directive on Official Languages for Communications and Services we should translate our query strings on the results page when the language switches. 

At the moment we do comply with the directive since it only mentions the Domain. 

https://www.tbs-sct.gc.ca/pol/doc-eng.aspx?id=26164#sec6.6

### PR DESCRIPTION
**Pull request by CalvinRodo**
## Description

Closes #366 
Make a query string that at least attempts to be okay in regards to the OLA.

Basically what this does is it takes the result object converts it to a string and base64 encodes it to pass on the URL. 

Then in the results page does the following  if the query string contains data it extracts the base64 encoded string,  decodes it and JSON.parses it. 

If the user futzes with the string we get a 500 error. 

## To test

1. navigate through the site
2. view the results page to ensure you get the results you expected and the query string is not in english or french.

### CODE REVIEW

File: routes/results/results.controller.js [hunk 1]
```diff
@@ -1,17 +1,23 @@
...
-  if (req.query) {
-    return req.query;
+  console.log(`queryString ${JSON.stringify(req.query)}`)
+  if (req.query === undefined || _.isEmpty(req.query)) {
+    return getSessionData(req);
+  }
+  try {
+    return JSON.parse(Buffer.from(req.query.q, 'base64').toString())
+  } catch (err) {
+    req.locals.logError(`Thrown error: ${JSON.stringify(err)} Invalid QueryString ${JSON.stringify(req.query)}`)
```
Comment (dsamojlenko) [inline routes/results/results.controller.js:18]: I think this is probably supposed to be `req.locals.log` ... when I trigger an error, I get an error message about this:

image

see:
image

File: app.js [hunk 2]
```diff
@@ -146,6 +147,9 @@ app.use(function (req, res, next) {
   next()
 })
 
+
+app.use(addLogger)
```
Comment (katedee) [inline app.js:151]: 👌

**Comment by dsamojlenko** [comment]
Did you mean to use `req.locals` throughout? Usually, things get attached to `res.locals`, there can be reason to attach to req, just wondering if this was deliberate?

**Comment by dsamojlenko** [comment]
On the question of structure, I've been meaning to move middleware out to a middleware folder for a while as well, so agree with that.

**Comment by dsamojlenko** [comment]
We've got Morgan logger installed, and being used for req logging ... would that be a fit for this?

File: routes/results/results.controller.js [hunk 3]
```diff
@@ -1,17 +1,23 @@
...
+const _ = require('lodash')
 
 const getData = (req) => {
   /**
    * If there's querystring data use it,
    * otherwise get it from the session.
    */
-  if (req.query) {
-    return req.query;
+  console.log(`queryString ${JSON.stringify(req.query)}`)
```
Comment (dsamojlenko) [inline routes/results/results.controller.js:11]: Did you mean to leave this console.log in place?

File: routes/results/results.controller.js (hunk 1 above)
Comment (CalvinRodo) [reply routes/results/results.controller.js:18]: 😬 Good catch, I originally had multiple loggers in there but removed them,

File: routes/results/results.controller.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (CalvinRodo) [reply routes/results/results.controller.js:11]: :+1:

Comment (CalvinRodo) [reply routes/results/results.controller.js:11]: No

**Comment by CalvinRodo** [comment]
@dsamojlenko Morgan is purely an http logger middleware, there is no way to explicitly call it, though I do like it's structured logging and so am trying to implement something simlar, I plan on fleshing out the logger to be a bit more fully featured with levels, and more session info for better tracability

File: routes/results/results.controller.js (hunk 1 above)
Comment (CalvinRodo) [reply routes/results/results.controller.js]: Fixed

File: routes/results/results.controller.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (CalvinRodo) [reply routes/results/results.controller.js]: Fixed

Comment (CalvinRodo) [reply routes/results/results.controller.js]: removed

**Pull request merged**